### ISSUE TITLE
Spyder says it's on version 5.4.0 even though updated to 5.4.3

### ISSUE THREAD

**Issue by JGSphaela** [spyder-ide/spyder#20875]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
I got this very strange problem: I installed spyder via pacman before, and I decided to remove it and installed a conda version instead. I remove the old spyder and install conda, which comes with a 5.4.3 spyder. `conda list` shows
```
spyder                    5.4.3           py310h06a4308_1
```
But when I start spyder, it still notify me to update to 5.4.3:
![image](https://user-images.githubusercontent.com/39881472/234880214-43587d8e-f4d3-4912-a4c2-292a4750e5cd.png)
And the version shows it's still on 5.3.0:
![image](https://user-images.githubusercontent.com/39881472/234880403-d2ea1487-8411-41c6-bde8-82568f03a610.png)

I removed anaconda and reinstall multiple times but the problem still exist. I also make sure that the spyder I started is from conda, by go in the `~/miniconda3/bin/` and start it directly, none of it helped. I'm really confused and don't know what to do now.

## Versions

* Spyder version: 5.3.0 None
* Python version: 3.10.10 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Linux 5.15.105-1-MANJARO

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<8.0.0       :  7.32.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.17.3 (OK)
keyring >=17.0.0              :  23.13.1 (OK)
nbconvert >=4.0               :  6.5.4 (OK)
numpydoc >=0.6.0              :  1.5.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0                :  2.16.2 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.4.1;<1.5.0          :  1.4.1 (OK)
pylsp_black >=1.2.0           :  1.2.0 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.2 (OK)
qtawesome >=1.0.2             :  1.2.2 (OK)
qtconsole >=5.3.0;<5.4.0      :  5.3.0 (OK)
qtpy >=2.0.1                  :  2.0.1 (OK)
rtree >=0.9.7                 :  1.0.1 (OK)
setuptools >=49.6.0           :  65.6.3 (OK)
sphinx >=0.6.6                :  5.0.2 (OK)
spyder_kernels >=2.3.0;<2.4.0 :  2.3.0 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
xdg >=0.26                    :  0.27 (OK)
zmq >=17                      :  23.2.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=3.0.0            :  None (NOK)
numpy >=1.7                   :  None (NOK)
pandas >=1.1.1                :  None (NOK)
scipy >=0.17.0                :  None (NOK)
sympy >=0.7.3                 :  None (NOK)
```

**Comment by dalthviz** [comment]
Hi @JGSphaela thank you for the feedback! Could you check what `sys.executable` returns inside Spyder and what it returns from the terminal where you used the `conda list`  command? So from Spyder, run in the IPython console something like:

```python
import sys
sys.exectuble
```

and from the cmd something like:

```
python -c "import sys; print(sys.executable)"
```

Are the results shown different in some way?

Not totally sure what could be happening :thinking but any new info in order to better understand this issue is greatly appreciated. Let us know!

**Comment by JGSphaela** [comment]
Hi @dalthviz, thanks for the reply! 'python -c "import sys; print(sys.executable)"' returns:
```
/home/user/miniconda3/bin/python
```
In the IPython console `sys.executable` return
```
sys.executable
Out[5]: '/home/user/miniconda3/bin/python'
```
If you want to know any more info just let me know!

**Comment by JGSphaela** [comment]
I don't know if this is related but I just found out my spyder can't open DataFrame in variable explorer and gives an error:
```
Traceback (most recent call last):
  File "/home/sphaela/.local/lib/python3.10/site-packages/spyder/plugins/variableexplorer/widgets/collectionsdelegate.py", line 203, in createEditor
    if not editor.setup_and_check(value, title=key):
  File "/home/sphaela/.local/lib/python3.10/site-packages/spyder/plugins/variableexplorer/widgets/dataframeeditor.py", line 946, in setup_and_check
    self.dataModel = DataFrameModel(data, parent=self)
  File "/home/sphaela/.local/lib/python3.10/site-packages/spyder/plugins/variableexplorer/widgets/dataframeeditor.py", line 135, in __init__
    self.max_min_col_update()
  File "/home/sphaela/.local/lib/python3.10/site-packages/spyder/plugins/variableexplorer/widgets/dataframeeditor.py", line 244, in max_min_col_update
    for __, col in self.df.iteritems():
  File "/home/sphaela/miniconda3/lib/python3.10/site-packages/pandas/core/generic.py", line 5989, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'iteritems'. Did you mean: 'isetitem'?
```

**Comment by dalthviz** [comment]
Thank you for the new info @JGSphaela ! The traceback you are seeing is related with https://github.com/spyder-ide/spyder/issues/20496 (which is an issue that was solved for Spyder 5.4.3).

However, something interesting is that the path shown related to Spyder doesn't point to your miniconda installation but instead to you `.local` directory in your user `home`  directory. Did you install Spyder in your machine before using pip with the `--user` flag? I think the problem comes from miniconda launching Spyder from your `~/.local` directory? What happens if you import spyder and check print spyder? Is the path shown pointing to `~/.local`?

If that is not the case I think you will need to remove the Spyder installation at `~/.local`

Let us know if the info above helps!

**Comment by JGSphaela** [comment]
Hi @dalthviz, that's exactly the cause of the problem! I didn't know I have the spyder installed with pip before, I removed it using `pip uninstall spyder` outside conda and now everything works! Thank you so much!
